### ISSUE TITLE
com.google.android.exoplayer2.source.UnrecognizedInputFormatException: None of the available extractors could read the stream

### ISSUE THREAD

**Issue by jerryisbusy** [google/ExoPlayer#7352]
hello, i have a video file here:
https://images2.bestjlb.com/jlbapp/5e09875981b2214e548f2eb09aac0e48/2020/05/v2jlbosscddf49b7d06c8d44b75253ec14b8987c(540x960)(7407ms).mp4 .

This video was recorded by an android phone and played well in the system album.
It can play well on Chrome browser and iPhone, Ubuntu Laptop,
but failed to play on any android phones through ExoPlayer (Version: 2.10.8).

Exception got here: 

```
05-09 17:50:42.505 24563 24563 W System.err: com.google.android.exoplayer2.ExoPlaybackException: com.google.android.exoplayer2.source.UnrecognizedInputFormatException: None of the available extractors (MatroskaExtractor, FragmentedMp4Extractor, Mp4Extractor, Mp3Extractor, AdtsExtractor, Ac3Extractor, TsExtractor, FlvExtractor, OggExtractor, PsExtractor, WavExtractor, AmrExtractor, Ac4Extractor) could read the stream.
05-09 17:50:42.505 24563 24563 W System.err: at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:386)
05-09 17:50:42.505 24563 24563 W System.err: at android.os.Handler.dispatchMessage(Handler.java:103)
05-09 17:50:42.505 24563 24563 W System.err: at android.os.Looper.loop(Looper.java:213)
05-09 17:50:42.505 24563 24563 W System.err: at android.os.HandlerThread.run(HandlerThread.java:67)
05-09 17:50:42.505 24563 24563 W System.err: Caused by: com.google.android.exoplayer2.source.UnrecognizedInputFormatException: None of the available extractors (MatroskaExtractor, FragmentedMp4Extractor, Mp4Extractor, Mp3Extractor, AdtsExtractor, Ac3Extractor, TsExtractor, FlvExtractor, OggExtractor, PsExtractor, WavExtractor, AmrExtractor, Ac4Extractor) could read the stream.
05-09 17:50:42.505 24563 24563 W System.err: at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractorHolder.selectExtractor(ProgressiveMediaPeriod.java:1074)
05-09 17:50:42.505 24563 24563 W System.err: at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:952)
05-09 17:50:42.505 24563 24563 W System.err: at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:394)
05-09 17:50:42.505 24563 24563 W System.err: at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
05-09 17:50:42.505 24563 24563 W System.err: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
05-09 17:50:42.505 24563 24563 W System.err: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
05-09 17:50:42.505 24563 24563 W System.err: at java.lang.Thread.run(Thread.java:929)
```

**Comment by krocard** [comment]
Unfortunately, I can not access the link you posted:

> https://images2.bestjlb.com/jlbapp/5e09875981b2214e548f2eb09aac0e48/2020/05/v2jlbosscddf49b7d06c8d44b75253ec14b8987c(540x960)(7407ms).mp4

403 Forbidden

Could you please post the specificity of the media you intend to play (bitrate, codecs, streams...)? You can get that information with mediainfo or ffprobe.

**Comment by jerryisbusy** [comment]
> Unfortunately, I can not access the link you posted:
> 
> > https://images2.bestjlb.com/jlbapp/5e09875981b2214e548f2eb09aac0e48/2020/05/v2jlbosscddf49b7d06c8d44b75253ec14b8987c(540x960)(7407ms).mp4
> 
> 403 Forbidden
> 
> Could you please post the specificity of the media you intend to play (bitrate, codecs, streams...)? You can get that information with mediainfo or ffprobe.

A 403 error returned while the video was first time opened in browser. But if you refresh the page, it will be fine. I just tried, and it works. I don't know why. 
By the way, use 'wget' to download it will be okay too.


Reveal the properties of this file, i got this:
Video:
Dimensions: 540 x 960
Codec: H.264 (Main Profile)
Framerate: 30 frames per second
Bitrate: 3515kbs

Audio:
Codec: MPEG-4 AAC
Channels: MONO
Sample Rate: 44100 Hz


ffprobe outputs:
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'v2jlbosscddf49b7d06c8d44b75253ec14b8987c(540x960)(7407ms).mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 1
    compatible_brands: mp41mp42isom
    creation_time   : 2019-12-12T06:13:34.000000Z
  Duration: 00:00:07.37, start: 0.000000, bitrate: 3646 kb/s
    Stream #0:0(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, mono, fltp, 96 kb/s (default)
    Metadata:
      creation_time   : 2019-12-12T06:13:34.000000Z
      handler_name    : Core Media Audio
    Stream #0:1(und): Video: h264 (Main) (avc1 / 0x31637661), yuv420p(tv, bt709), 540x960, 5300 kb/s, 30 fps, 30 tbr, 600 tbn, 1200 tbc (default)
    Metadata:
      creation_time   : 2019-12-12T06:13:34.000000Z
      handler_name    : Core Media Video

**Comment by krocard** [comment]
Testing your content with our demo app from our development branch (1), it ExoPlayer can play it without issue with it's `Mp4Extractor`).

I'll try again from 2.10.8, the version you are using.

(1)` adb shell 'am start -a com.google.android.exoplayer.demo.action.VIEW -d "https://images2.bestjlb.com/jlbapp/5e09875981b2214e548f2eb09aac0e48/2020/05/v2jlbosscddf49b7d06c8d44b75253ec14b8987c(540x960)(7407ms).mp4"'`

Note that towards the end of the playback playback fails due to an error 416, which indicates the file is truncated, but this is unrelated as your issue is an `UnrecognizedInputFormatException`.

**Comment by krocard** [comment]
I can confirm the issue existed on 2.10.8 but is fixed on the latest version 2.11.4. Please update your version of ExoPlayer to the latest release.

**Comment by jerryisbusy** [comment]
> I can confirm the issue existed on 2.10.8 but is fixed on the latest version 2.11.4. Please update your version of ExoPlayer to the latest release.

Thanks. I will update to the latest release.
